client/src/main/java/org/ehrbase/client/openehrclient/defaultrestclient/DefaultRestAqlEndpoint.java

```diff
@@ -178,6 +179,120 @@ public QueryResponseData executeRaw(Query query, ParameterValue... parameters) {
         }
     }
 
+    @Override
+    public QueryResponseData executeStoredQuery(
+            String qualifiedQueryName,
+            String version,
+            // Optional<UUID> ehrId, // TODO: ehr_id not available in openEHR 1.0.0 spec. Has to be included in
+            // queryParameter
+            Optional<Integer> offset,
+            Optional<Integer> fetch,
+            Optional<String> queryParameters) {
```

Comment:
I would prefer to have a parameter object here.

Reply:
suggestion. use a dto with builder Patten.  then in the its clear what is required and what is optional . and the returns can then return optinal for the optional values

Reply:
parameter object added